ocaml/lambda/translprim.ml

```diff
@@ -1059,7 +1059,7 @@ let specialize_primitive env loc ty ~has_constant_constructor prim =
                                   Jkind.Sort.for_block_element typ))
           fields
       in
-      let useful = List.exists (fun knd -> knd <> Pgenval) shape in
+      let useful = List.exists (fun knd -> knd.raw_kind <> Pgenval) shape in
```

Comment:
Wouldn't `useful` now be `List.exists (fun knd -> knd <> Lambda.generic_value) shape` instead?

Reply:
It's not completely clear that knowing that a `Pgenval` is not `Null` is useful, but your suggestion is also reasonable and avoids having to answer this complicated question.